DataServer/portAllocator.py

```diff
@@ -28,22 +49,41 @@ def reservePort(self) -> int:
         Returns:
             int: Next available port number
         """
-        
-        # TODO reimplement this
-        self.__current += 1
-        if self.__current > self.__end:
-            raise RuntimeError
-        return self.__current
+        self.createDict(self.__start, self.__end)
```

Comment:
```
>       self.createDict(self.__start, self.__end)
E       TypeError: createDict() takes 1 positional argument but 3 were given

DataServer/portAllocator.py:52: TypeError
```

Reply:
This also seems excessive - why do you need to test all of the ports every time we ask for a new port?